edx_rest_api_client/auth.py

```diff
@@ -48,3 +48,16 @@ def __call__(self, r):
         """ Update the request headers. """
         r.headers['Authorization'] = 'Bearer {}'.format(self.token)
         return r
+
+
+class OidcIdTokenAuth(AuthBase):
```

Comment:
@clintonb that class has some other required parameters `username`, `full_name`, `email`, `signing_key`. Does it make sense to keep this as separate class?